pandas/core/indexing.py

```diff
@@ -2268,6 +2269,69 @@ def convert_to_index_sliceable(obj, key):
     return None
 
 
+def check_bool_array_indexer(array: AnyArrayLike, mask: AnyArrayLike) -> np.ndarray:
```

Comment:
i would like to see these utilities split out of here (pandas.core.indexing is huge). but can certainly be done later.

Reply:
this is also simple enough it could go into core.indexers